ENH: Download data from test server to protected DB

turanoo opened the pull request:
Fixes #14

turanoo committed:
Keeping Track

turanoo committed:
ENH: Download data from the test server to the protected DB sheet

Fixes #14

thermokarst commented on src/utils.js:
```diff
@@ -6,14 +6,14 @@ export function onOpen() {
-  if (obj.msg === 'hello world') {
-    rangeList.setValue(obj.msg);
-  }
+  const DB = SpreadsheetApp.getActiveSpreadsheet().getSheetByName('DB');
```
We need to guard against the case where this sheet doesn't exist. Check out this error that comes up when the 'DB' sheet isn't present:

```javascript
TypeError: Cannot call method "getRange" of null.
```

thermokarst commented on src/utils.js:
```diff
@@ -6,14 +6,14 @@ export function onOpen() {
+  const DB = SpreadsheetApp.getActiveSpreadsheet().getSheetByName('DB');
+  const fetchResponse = UrlFetchApp.fetch('http://ghost.mggen.nau.edu:8081/basic/csv/lite');
+  const text = fetchResponse.getContentText();
+  const valuesArr = text.split('\n').map(e => e.split(','));
```
It seems like this function (`getData`) is doing two different things: fetching data, then parsing data. Maybe it makes sense to separate into two (or more) functions with discrete responsibilities?

thermokarst replied:
The reason I commented on this line (12) is because this looks like the first line related to parsing data.

thermokarst reviewed (changes requested):
Thanks @turanoo, comments inline.

turanoo committed:
SQUASH: Responding to @thermokarst's review

When fetching data to a protected DB sheet we first check if the sheet exists.
Simplified getData function into two methods (fetchData and parseData).
Changed ordering of methods in utils.js
Changed some variable names to ensure consistency.
Edited the main.js file to reflect the correct order of functions declared.

turanoo committed:
SQUASH: Created a separate function for posting data

turanoo committed:
SQUASH: Pair programming with @thermokarst

turanoo committed:
SQUASH: Pairprogramming with @thermokarst

turanoo committed:
SQUASH: Pairprogramming with @thermokarst

turanoo committed:
SQUASH: Cleaned up exporting functions

turanoo committed:
SQUASH: Used => syntax on some functions

turanoo committed:
SQUASH: MISC Cleaning up

turanoo committed:
SQUASH: Added the original yarn.lock without any updates

turanoo committed:
SQUASH: Changed utils.js to util.js to match the file naming convention

thermokarst reviewed (approved):
Group-reviewed with @turanoo. Thanks, this looks awesome! 🎉
